fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/Index.java

```diff
@@ -646,12 +696,13 @@ public boolean equals(Object o) {
                 && this.addedVersion == that.addedVersion
                 && this.lastModifiedVersion == that.lastModifiedVersion
                 && Arrays.equals(this.primaryKeyComponentPositions, that.primaryKeyComponentPositions)
-                && this.options.equals(that.options);
+                && this.options.equals(that.options)
+                && (Objects.equals(this.predicate, that.predicate));
     }
 
     @Override
     public int hashCode() {
         // Within the context of a single RecordMetaData, this should be sufficient
```

Comment:
Why isn't this still true? If you had a `Map` from `Index`, presumably all from the same `RecordMetaData`, wouldn't this be good enough?

Reply:
I thought about it again, you're right, it is actually still true even with the addition of the predicate. I will restore it.